tox/_quickstart.py

```diff
@@ -110,186 +106,144 @@ def ok(x):
-    - pytest
-    - python setup.py test
-    - nosetests package.module
-    - trial package.module''')
-    do_prompt(d, 'commands', 'Command to run to test project', '{envpython} setup.py test')
-
-    default_deps = ' '
-    if any(c in d['commands'] for c in ['pytest', 'py.test']):
-        default_deps = 'pytest'
-    if 'nosetests' in d['commands']:
-        default_deps = 'nose'
-    if 'trial' in d['commands']:
-        default_deps = 'twisted'
-
-    print('''
-What extra dependencies do your tests have?''')
-    do_prompt(d, 'deps', 'Comma-separated list of dependencies', default_deps)
-
-
-def process_input(d):
-    d['envlist'] = ', '.join([env for env in all_envs if d.get(env) is True])
-    d['deps'] = '\n' + '\n'.join([
-        '    %s' % dep.strip()
-        for dep in d['deps'].split(',')])
-
-    return d
-
-
-def rtrim_right(text):
-    lines = []
-    for line in text.split("\n"):
-        lines.append(line.rstrip())
-    return "\n".join(lines)
-
-
-def generate(d, overwrite=True, silent=False):
+    map_[key] = modificator(answer, map_.get(key)) if modificator else answer
+
+
+def ask_user(map_):
+    """modify *map_* in place by getting info from the user."""
+    print('Welcome to the tox %s quickstart utility.' % tox.__version__)
+    print('This utility will ask you a few questions and then generate a simple configuration '
+          'file to help get you started using tox.\n'
+          'Please enter values for the following settings (just press Enter to accept a '
+          'default value, if one is given in brackets).\n')
+    print('What Python versions do you want to test against?\n'
+          '    [1] %s\n'
+          '    [2] py27, %s\n'
+          '    [3] (All versions) %s\n'
+          '    [4] Choose each one-by-one' % (
+              CURRENT_RELEASE_ENV, CURRENT_RELEASE_ENV, ', '.join(ALL_PY_ENVS)))
+    do_prompt(map_, 'canned_pyenvs', 'Enter the number of your choice',
+              default='3', validator=choice('1', '2', '3', '4'))
+    if map_['canned_pyenvs'] == '1':
+        map_[CURRENT_RELEASE_ENV] = True
+    elif map_['canned_pyenvs'] == '2':
+        for pyenv in ('py27', CURRENT_RELEASE_ENV):
+            map_[pyenv] = True
+    elif map_['canned_pyenvs'] == '3':
+        for pyenv in ALL_PY_ENVS:
+            map_[pyenv] = True
+    elif map_['canned_pyenvs'] == '4':
+        for pyenv in ALL_PY_ENVS:
+            if pyenv not in map_:
+                do_prompt(map_, pyenv, 'Test your project with %s (Y/n)' % pyenv, 'Y',
+                          validator=boolean)
+    print('What command should be used to test your project? Examples:\n'
+          '    - pytest\n'
+          '    - python setup.py test\n'
+          '    - nosetests package.module\n'
+          '    - trial package.module\n')
+    do_prompt(map_, 'commands', 'Type the command to run your tests',
+              default='pytest', modificator=list_modificator)
```

Comment:
Personally prefer to keep defaults/recommendations to the standard library. Maybe `python -m unittest discover`?

Reply:
Good idea - I also thought that nose is not really alive and well anymore for a long time, so I will replace it with your suggested unittest call.